runtime/src/main/scala/akka/grpc/internal/GrpcResponseHelpers.scala

```diff
@@ -39,6 +39,22 @@ object GrpcResponseHelpers {
       system: ClassicActorSystemProvider): HttpResponse =
     GrpcResponseHelpers(e, Source.single(GrpcEntityHelpers.trailer(Status.OK)), eHandler)
 
+  def responseForSingleElement[T](e: T, eHandler: ActorSystem => PartialFunction[Throwable, Trailers])(
+      implicit m: ProtobufSerializer[T],
+      writer: GrpcProtocolWriter,
+      system: ClassicActorSystemProvider): HttpResponse = {
+    def x = eHandler
+    def f = system
+    if (false) { x; f }
+
+    val data = DataFrame(m.serialize(e))
+    val strictEntity = HttpEntity.Strict(writer.contentType, writer.encodeFrame(data).data)
+    val trailer = Trailer(GrpcEntityHelpers.trailer(Status.OK).trailers)
+    HttpResponse(
+      headers = immutable.Seq(headers.`Message-Encoding`(writer.messageEncoding.name)),
+      entity = strictEntity).addAttribute(AttributeKeys.trailer, trailer)
```

Comment:
Is it possible to add attributes in the HttpResponse constructor to avoid having to potentially allocate a new HttpResponse immediately?